### ISSUE TITLE
prefer-flat-map should not trigger with .map().flat(Infinity)

### ISSUE THREAD

**Issue by Sembiance** [sindresorhus/eslint-plugin-unicorn#848]
The prefer-flat-map check is already quiet if a number >1 is passed in such as:
```javascript
a.map(v => v).flat(999); // Correct behavior, no error
```

This is because .flatMap() can only flatten once.

However the following code DOES trigger the error:
```javascript
a.map(v => v).flat(Infinity); // Incorrect behavior, error triggered
```

Should be as simple as changing https://github.com/sindresorhus/eslint-plugin-unicorn/blob/6c9b53191af7534aad2fa3b72f943e5cd0c4374c/rules/prefer-flat-map.js#L142-L148 to:
```javascript
	if (
		node.arguments.length === 1 &&
		((node.arguments[0].type === 'Literal' &&
		  node.arguments[0].value !== 1) ||
	 	 (node.arguments[0].type === 'Identifier' &&
		  node.arguments[0].value === 'Infinity'))
	) {
		return;
	}
```

**Comment by William-Schutte** [comment]
Hey, found this through Hacktober. Seems simple enough. Do I need to be assigned to do a pull request?
